---
title: "targeted_antibiotic_visualization_statistics"
author: "Reilly Cooper"
date: "10/31/2019"
output: html_document
---

Loading necessary packages and functions.
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(phyloseq)
library(tidyverse)
library(cowplot)
library(grid)
library(gridExtra)
library(microbiome)
library(vegan)
library(knitr)
library(survminer)
library(survival)
library(DESeq2)
library(data.table)
library(multcompView)
library(survival)
library(survminer)
library(wesanderson)
library(cowplot)
theme_set(theme_bw())
pal <- c("Alphaproteobacteria" = "#9A872D", "-" = "#AEA8A8", "Flavobacteriia" = "#F5CDB6", "Gammaproteobacteria" = "#F7B0AA", 
                               "Betaproteobacteria" = "#FDDDA4", "Sphingobacteriia" = "#76A08A", "Cytophagia" = "#C7CEF6", "Verrucomicrobiae" = "#7496D2",
                               "Actinobacteria" = "#C4CFD0", "Bacilli" = "#456355", "Clostridia" = "#B5966D", "Deinococci" = "#CECD7B", "Planctomycetia" = "#F8DF4F")
# Function to make Tukey HSD labels for figures.
generate_label_df <- function(TUKEY, variable){
  
  # Extract labels and factor levels from Tukey post-hoc 
  Tukey.levels <- TUKEY[[variable]][,4]
  Tukey.labels <- data.frame(multcompLetters(Tukey.levels)['Letters'])
  
  #I need to put the labels in the same order as in the boxplot :
  Tukey.labels$treatment=rownames(Tukey.labels)
  Tukey.labels=Tukey.labels[order(Tukey.labels$treatment) , ]
  return(Tukey.labels)
}

# Function to filter 16S read counts of 0 and convert counts to log scale for DESeq2
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
```

Loading all data.
```{r}
# Cumulative Daphnia reproduction
reproSum <- read.csv("reprodsum.csv")
# First brood data
firstBrood <- read.csv("repro_first_brood.csv")
# Daphnia survival for statistics
survivalStats <- read.csv("formatteddeath.csv")
# Daphnia survival for visualization
survivalViz <- read.csv("survival_death.csv")
# Daphnia growth
growth <- read.csv("bodysize.csv")
# Daphnia microbiome data
microbiome <- readRDS("rcClean.rds")
```

Visualization and statistics of cumulative reproduction. Because only one individual in the cold, 11C treatment had offspring, I am subsetting the data to just the 19C treatment for visualization.
```{r cumulative reproduction}
# ANOVA to test effects of antibiotics, temperature, and the interaction between the two
repro.anova = aov(repro_sum ~ ab_trt*temp_trt, reproSum)
summary(repro.anova)
# Tukey HSD post-hoc comparisons to find significantly different treatments.
tHSD <- TukeyHSD(repro.anova)
# Reording factor levels for clearer figure
reproSum$ab_trt <- factor(reproSum$ab_trt,levels = c("NONE","AZT", "ERY", "SFX", "ALL"))
reproSubset <- subset(reproSum, temp_trt == "19C")
# Using the generate_label_df function, creating letter labels of significantly different groups to add to a figure.
labels<-generate_label_df(tHSD , "ab_trt")
names(labels)<-c('Letters','ab_trt')
yvalue<-aggregate(.~ab_trt, data=reproSubset, mean)
final<-merge(labels,yvalue)
# Generating figure
cumulative_reproduction <- ggplot(reproSubset, aes(ab_trt, repro_sum)) + geom_boxplot(outlier.shape = NA) + 
  geom_jitter(width = .1, size = .5, alpha = .2) + xlab("Antibiotic Treatment") + ylab("Cumulative Reproduction") + 
  geom_text(data = final, aes(x = ab_trt, y = repro_sum, label = Letters),vjust=-8,hjust=-1, size = 4, fontface = "bold")
cumulative_reproduction
# Saving figure
ggsave("cumulative_reproduction.pdf", cumulative_reproduction, units = "in", width = 4, height = 4, dpi = 300)
```

Visualization and statistics of first brood timing. Again, subsetting to only the 19C treatment.
```{r first brood timing}
# ANOVA to test for differences in timing
first.anova = aov(day ~ ab, data = firstBrood)
summary(first.anova)
# Tukey HSD comparisons
tHSD.first <- TukeyHSD(first.anova)
# Reordering factor levels for clearer figure
firstBrood$ab <- factor(firstBrood$ab,levels = c("NONE","AZT", "ERY", "SFX", "ALL"))
# Using the generate_label_df function, creating letter labels of significantly different groups to add to a figure.
labels<-generate_label_df(tHSD.first , "ab")
names(labels)<-c('Letters','ab')
yvalue<-aggregate(.~ab, data=firstBrood, mean)
final<-merge(labels,yvalue)
# Generating figure
day_of_firstbrood <- ggplot(firstBrood, aes(ab, day)) + stat_summary(fun.data = "mean_cl_normal", color = "#9A872D") +
  geom_jitter(width = .1, size = .5, alpha = .5, height = 0) +
  xlab("Antibiotic Treatment") + ylab("Day of First Brood") + ylim(10,21) + 
  geom_text(data = final, aes(x = ab, y = day, label = Letters),vjust=-8,hjust=-1, size = 4, fontface = "bold")
# Saving figure
ggsave("day_of_first_brood.pdf", day_of_firstbrood, units = "in", width = 4, height = 4, dpi = 300)
```

Visualization and statistics of Daphnia growth.
```{r growth}
# Reorder levels to make graphs nice
growth$ab_trt <- factor(growth$ab_trt,levels = c("NONE","AZT", "ERY", "SFX", "ALL"))
growth$temp_trt <- factor(growth$temp_trt,levels = c("19C", "11C"))
# Add column of growth (size_end - size_begin)
growth <- mutate(growth, growth = size_end - size_begin)
# Statistics to test for the effects of antibiotics, temperature, and the interaction
growth.anova = aov(growth ~ ab_trt*temp_trt, data = growth)
summary(growth.anova)
tHSD.growth <- TukeyHSD(growth.anova)
tHSD.growth
# Making figure. No added significance labels here, would get too cluttered.
growth_by_treatment <- ggplot(growth, (aes(ab_trt, growth))) + 
  facet_wrap(~temp_trt) + geom_boxplot(outlier.shape = NA) + 
  geom_jitter(width = .1, size = .5, alpha = .5, height = 0) + 
  xlab("Antibiotic Treatment") + ylab("Growth (mm)") +
  theme(strip.background =element_rect(fill="white"))
# Saving figure
ggsave("growth_by_treatment.pdf", growth_by_treatment, units = "in", width = 4, height = 4, dpi = 300)
```

Visualization and statistics of Daphnia survival.
```{r survival}
# Releveling factors so comparisons are against control temperature and no antibiotics
survivalStats$temp_trt = relevel(survivalStats$temp_trt, ref = "19C")
survivalStats$ab_trt <- factor(survivalStats$ab_trt,levels = c("NONE","AZT", "ERY", "SFX", "ALL"))
# Renaming columns to get nicer outputs
colnames(survivalStats) <- c("Individual", "Antibiotic_Treatment", "Temperature_Treatment", "status", "time")
# Comparing survival across all treatments
survivalObject <- Surv(time = survivalStats$time, event = survivalStats$status)
fit <- coxph(survivalObject ~ Antibiotic_Treatment + Temperature_Treatment, data = survivalStats)
summary(fit)
# Visualizing hazard ratios
survival_hazard_ratios <- ggforest(fit, data = survivalStats, fontsize = .5)
# Saving hazard ratio visualization
ggsave("survival_hazard_ratios.pdf", survival_hazard_ratios, units = "in", width = 8.5, height = 11, dpi = 300)

# Cleaning some empty data for visualization
survivalViz <- survivalViz[0:210,0:5]
# Reorder levels for nice graphs
survivalViz$ab_trt <- factor(survivalViz$ab_trt,levels = c("NONE","AZT", "ERY", "SFX", "ALL"))
survivalViz$temp_trt <- factor(survivalViz$temp_trt,levels = c("19C", "11C"))
# Make figure
survival_curves <- ggplot(survivalViz, aes(day, propalive, group = ab_trt, linetype = ab_trt, color = ab_trt)) + 
  facet_wrap(~temp_trt) + geom_line(size = 1) +
  theme(strip.background =element_rect(fill="white")) + xlab("Day") + ylab("Proportion Alive") + 
  scale_color_manual(values = c("NONE" = "#9A872D", "AZT" = "#F5CDB6", "ERY" = "#F7B0AA", "SFX" = "#FDDDA4", "ALL" = "#76A08A")) +
  guides(fill = guide_legend(title = "Antibiotic Treatment"))
# Saving figure - changing legend title to "Antibiotic Treatment" using Adobe Illustrator after
ggsave("survival_curves.pdf", survival_curves, units = "in", width = 6, height = 6, dpi = 300)
```

Visualizing the most abundant taxa present in the adult Daphnia microbiome in standard conditions. Metadata has temperature treatments as "18C" and "11C"; "18C" == "19C". No statistics are being done in this part.
```{r adult microbiome}
# Removing sample with no reads, the first round of data that wasn't viable, and any reads associating with Chlamydomonas
microbiome = subset_samples(microbiome, Sample != "E2T02S3")
microbiome = subset_samples(microbiome, experiment == "two")
microbiome = subset_taxa(microbiome, Phylum != "Cyanobacteria/Chloroplast")
# Selecting samples from adults in regular temperatures and no antibiotics
adultMicrobiome = subset_samples(microbiome, temp_treatment == "18C" & antibiotic_treatment == "NONE")
# Selecting the top 10 ASVs
TopASVs = names(sort(taxa_sums(adultMicrobiome), TRUE)[1:10])
top10 = prune_taxa(TopASVs, adultMicrobiome)
# Merging samples to create single bar
adultMicrobiome.merged = merge_samples(top10, "antibiotic_treatment")
# Transforming sample counts into percentages
adultMicrobiome.transformed = transform_sample_counts(adultMicrobiome.merged, function(x) x/sum(x))
# Conglomerating at the genus level to make figure cleaner
adultMicrobiome.glommed = tax_glom(adultMicrobiome.transformed, "Genus")
# Visualizing at the genus level
adultMicrobiomeGenera = plot_bar(adultMicrobiome.glommed, "Sample", fill = "Genus") + scale_fill_manual(values = c("Acidovorax" = "#456355", "Fluviicola" = "#AC6E49", "Limnohabitans" = "#957A6D",
                                                                                                                   "Pedobacter" = "#CB9E23", "Prosthecobacter" = "#541F12", "Pseudomonas" = "#A35E60",
                                                                                                                   "Vitreoscilla" = "#DAECED"))
# Selecting for all classes with abundances >0.01 
adultMicrobiome.mergedUnfiltered = merge_samples(adultMicrobiome, "antibiotic_treatment")
adultMicrobiome.unfilteredTransformed = transform_sample_counts(adultMicrobiome.mergedUnfiltered, function(x) x/sum(x))
adultMicrobiome.abundanceFiltered = filter_taxa(adultMicrobiome.unfilteredTransformed, function(x) mean(x) > .01, TRUE)
adultMicrobiome.classGlommed = tax_glom(adultMicrobiome.abundanceFiltered, "Class")
# Visualizing
adultMicrobiomeClass = plot_bar(adultMicrobiome.classGlommed, "Sample", fill = "Class") +
  scale_fill_manual(values = pal)
# Arranging genus-level and class-level figures into one figure.
adultMicrobiomeFigure = arrangeGrob(adultMicrobiomeGenera, adultMicrobiomeClass, nrow = 1)
# Saving image - will remove "NONE" and "Sample" from X axis in Adobe Illustrator
ggsave("standard_adult_microbiome.pdf", adultMicrobiomeFigure, units = "in", height = 8, width = 8, dpi = 300)
```

Statistics and visualization for differences among antibiotic treatments in regular environmental conditions. Using a PERMANOVA to calculate differences among treatments. Then visualizing the differences in microbiome composition in each temperature.
```{r antibiotics microbiome}
# To do PERMANOVA, need to get pariwise comparisons.
# Creating lists to pull out pairwise comparisons against NONE.
noneAzt = c("NONE", "AZT")
noneEry = c("NONE", "ERY")
noneSfx = c("NONE", "SFX")
noneAll = c("NONE", "ALL")
# Subsetting microbiome data based on pairs
noneVazt = subset_samples(microbiome, antibiotic_treatment %in% noneAzt)
noneVery = subset_samples(microbiome, antibiotic_treatment %in% noneEry)
noneVsfx = subset_samples(microbiome, antibiotic_treatment %in% noneSfx)
noneVall = subset_samples(microbiome, antibiotic_treatment %in% noneAll)
# Transforming all sample counts to proportions
noneVaztTransformed = transform_sample_counts(noneVazt, function(x) x/sum(x))
nonveVeryTransformed = transform_sample_counts(noneVery, function(x) x/sum(x))
noneVsfxTransformed = transform_sample_counts(noneVsfx, function(x) x/sum(x))
noneVallTransformed = transform_sample_counts(noneVall, function(x) x/sum(x))
# Pulling out ASV data and metadata
noneVaztASV <- abundances(noneVaztTransformed)
noneVeryASV <- abundances(nonveVeryTransformed)
noneVsfxASV <- abundances(noneVsfxTransformed)
noneVallASV <- abundances(noneVallTransformed)
noneVaztMeta <- meta(noneVaztTransformed)
noneVeryMeta <- meta(nonveVeryTransformed)
noneVsfxMeta <- meta(noneVsfxTransformed)
noneVallMeta <- meta(noneVallTransformed)
# Performing pairwise comparisons
noneVaztPERM <- adonis(t(noneVaztASV) ~ antibiotic_treatment, data = noneVaztMeta, permutation = 999, method = "bray")
noneVeryPERM <- adonis(t(noneVeryASV) ~ antibiotic_treatment, data = noneVeryMeta, permutation = 999, method = "bray")
noneVsfxPERM <- adonis(t(noneVsfxASV) ~ antibiotic_treatment, data = noneVsfxMeta, permutation = 999, method = "bray")
noneVallPERM <- adonis(t(noneVallASV) ~ antibiotic_treatment, data = noneVallMeta, permutation = 999, method = "bray")
# Pulling out relevant statistics from adonis object
noneVaztResults <- noneVaztPERM$aov.tab
noneVeryResults <- noneVeryPERM$aov.tab
noneVsfxResults <- noneVsfxPERM$aov.tab
noneVallResults <- noneVallPERM$aov.tab
noneVaztResults %>% print()
noneVeryResults %>% print()
noneVsfxResults %>% print()
noneVallResults %>% print()

# Visualizing
microbiomeABonly = subset_samples(microbiome, temp_treatment == "18C")
# Merge samples to create single bar for each antibiotic treatment
microbiomeABmerged = merge_samples(microbiomeABonly, "antibiotic_treatment")
# Transform counts to proportions
microbiomeABtransformed = transform_sample_counts(microbiomeABmerged, function(x) x/sum(x))
# Conglomerate at the class level
microbiomeABclass = tax_glom(microbiomeABtransformed, "Class")
# Renaming factor levels - merging samples turns factors into numbers, so need to turn back to names
sample_data(microbiomeABclass)$antibiotic_treatment <- c("ALL", "AZT", "ERY", "NONE", "SFX")
# Reordering levels for consistency across figures
microbiomeABclass@sam_data$antibiotic_treatment <- factor(microbiomeABclass@sam_data$antibiotic_treatment, levels = c("NONE","AZT","ERY","SFX","ALL"))
# Making figure
microbiomeByAB = plot_bar(microbiomeABclass, "antibiotic_treatment", fill = "Class") + theme(legend.position = "none", axis.ticks.x = element_blank()) +
  scale_fill_manual(values = pal)
# Saving figure - will remove "antibiotic_treatment" from figure and change "-" to "Other"

microbiomecoldABonly = subset_samples(microbiome, temp_treatment == "11C")
microbiomecoldABmerged = merge_samples(microbiomecoldABonly, "antibiotic_treatment")
microbiomecoldABtransformed = transform_sample_counts(microbiomecoldABmerged, function(x) x/sum(x))
microbiomecoldABclass = tax_glom(microbiomecoldABtransformed, "Class")
sample_data(microbiomecoldABclass)$antibiotic_treatment <- c("ALL", "AZT", "ERY", "NONE", "SFX")
microbiomecoldABclass@sam_data$antibiotic_treatment <- factor(microbiomecoldABclass@sam_data$antibiotic_treatment, levels = c("NONE","AZT","ERY","SFX","ALL"))
microbiomecoldByAB = plot_bar(microbiomecoldABclass, "antibiotic_treatment", fill = "Class") + theme( legend.position = "none", axis.ticks.x = element_blank()) +
  scale_fill_manual(values = pal)

# Combining figures
microbiome_across_temps = plot_grid(microbiomeByAB, microbiomecoldByAB, nrow = 1)
# Generating and saving legend
for_ab_temp_legend = plot_bar(microbiomeABclass, "antibiotic_treatment", fill = "Class") + guides(fill = guide_legend(title = "Class")) +
  theme(axis.text.x = element_text(angle = 0, hjust = .5), legend.position = "bottom") + 
  scale_fill_manual(values = pal)
ab_temp_legend <- get_legend(for_ab_temp_legend)

# Saving both
ggsave("microbiome_across_temps_antibiotics.pdf", microbiome_across_temps, units = "in", height = 8, width = 8, dpi = 300)
ggsave("microbiome_across_temps_antibiotics_legend.pdf", ab_temp_legend, units = "in", height = 8, width = 8, dpi = 300)
```

Statistics for differences in composition between temperature treatments across antibiotic treatments.
```{r temperature microbiome}
# Subsetting by antibiotic treatment
NONEtrt = subset_samples(microbiome, antibiotic_treatment == "NONE")
AZTtrt = subset_samples(microbiome, antibiotic_treatment == "AZT")
ERYtrt = subset_samples(microbiome, antibiotic_treatment == "ERY")
SFXtrt = subset_samples(microbiome, antibiotic_treatment == "SFX")
ALLtrt = subset_samples(microbiome, antibiotic_treatment == "ALL")
# Transforming sample counts to proportions
NONEtransformed = transform_sample_counts(NONEtrt, function(x) x/sum(x))
AZTtransformed = transform_sample_counts(AZTtrt, function(x) x/sum(x))
ERYtransformed = transform_sample_counts(ERYtrt, function(x) x/sum(x))
SFXtransformed = transform_sample_counts(SFXtrt, function(x) x/sum(x))
ALLtransformed = transform_sample_counts(ALLtrt, function(x) x/sum(x))
# Pulling out ASV data and metadata
NONEasvs <- abundances(NONEtransformed)
AZTasvs <- abundances(AZTtransformed)
ERYasvs <- abundances(ERYtransformed)
SFXasvs <- abundances(SFXtransformed)
ALLasvs <- abundances(ALLtransformed)
NONEmeta <- meta(NONEtransformed)
AZTmeta <- meta(AZTtransformed)
ERYmeta <- meta(ERYtransformed)
SFXmeta <- meta(SFXtransformed)
ALLmeta <- meta(ALLtransformed)
# Pairwise PERMANOVAs
NONEperm <- adonis(t(NONEasvs) ~ temp_treatment, data = NONEmeta, permutation = 999, method = "bray")
AZTperm <- adonis(t(AZTasvs) ~ temp_treatment, data = AZTmeta, permutation = 999, method = "bray")
ERYperm <- adonis(t(ERYasvs) ~ temp_treatment, data = ERYmeta, permutation = 999, method = "bray")
SFXperm <- adonis(t(SFXasvs) ~ temp_treatment, data = SFXmeta, permutation = 999, method = "bray")
ALLperm <- adonis(t(ALLasvs) ~ temp_treatment, data = ALLmeta, permutation = 999, method = "bray")
# Pulling relevant statistics out
NONEresults <- NONEperm$aov.tab %>% print()
AZTresults <- AZTperm$aov.tab %>% print()
ERYresults <- ERYperm$aov.tab %>% print()
SFXresults <- SFXperm$aov.tab %>% print()
ALLresults <- ALLperm$aov.tab %>% print()
```


Prepping data for visualization and statistics of differentially abundant taxa across treatments using DESeq2.
```{r deseq}
# Creating DESeq objects for antibiotic treatments and temperature treatments
# Across antibiotics
deseqABonly = phyloseq_to_deseq2(microbiomeABonly, ~ antibiotic_treatment)

# For each antibiotic across temperatures
deseqTEMP.NONE = phyloseq_to_deseq2(NONEtrt, ~ temp_treatment)
deseqTEMP.AZT = phyloseq_to_deseq2(AZTtrt, ~ temp_treatment) 
deseqTEMP.ERY = phyloseq_to_deseq2(ERYtrt, ~ temp_treatment)
deseqTEMP.SFX = phyloseq_to_deseq2(SFXtrt, ~ temp_treatment)
deseqTEMP.ALL = phyloseq_to_deseq2(ALLtrt, ~ temp_treatment)

# Transforming counts to DESeq-readable distribution
geoMeansAB = apply(counts(deseqABonly), 1, gm_mean)
geoMeansTEMP.NONE = apply(counts(deseqTEMP.NONE), 1, gm_mean)
geoMeansTEMP.AZT = apply(counts(deseqTEMP.AZT), 1, gm_mean)
geoMeansTEMP.ERY = apply(counts(deseqTEMP.ERY), 1, gm_mean)
geoMeansTEMP.SFX = apply(counts(deseqTEMP.SFX), 1, gm_mean)
geoMeansTEMP.ALL = apply(counts(deseqTEMP.ALL), 1, gm_mean)

# Estimating size factors for AB/Temp. Size factors are a way to estimate sampling depth of a library, ensuring that DESeq2 is able to accurately assess differential abundance even if only a few ASVs dominate samples.
deseqABonly = estimateSizeFactors(deseqABonly, geoMeans = geoMeansAB)
deseqTEMP.NONE = estimateSizeFactors(deseqTEMP.NONE, geoMeans = geoMeansTEMP.NONE)
deseqTEMP.AZT = estimateSizeFactors(deseqTEMP.AZT, geoMeans = geoMeansTEMP.AZT)
deseqTEMP.ERY = estimateSizeFactors(deseqTEMP.ERY, geoMeans = geoMeansTEMP.ERY)
deseqTEMP.SFX = estimateSizeFactors(deseqTEMP.SFX, geoMeans = geoMeansTEMP.SFX)
deseqTEMP.ALL = estimateSizeFactors(deseqTEMP.ALL, geoMeans = geoMeansTEMP.ALL)

# Calculating differential abundances
deseqABonly = DESeq(deseqABonly, fitType = "local")
deseqTEMP.NONE = DESeq(deseqTEMP.NONE, fitType = "local")
deseqTEMP.AZT = DESeq(deseqTEMP.AZT, fitType = "local")
deseqTEMP.ERY = DESeq(deseqTEMP.ERY, fitType = "local")
deseqTEMP.SFX = DESeq(deseqTEMP.SFX, fitType = "local")
deseqTEMP.ALL = DESeq(deseqTEMP.ALL, fitType = "local")
```

First, looking at differentially abundant taxa across antibiotics in control conditions.
```{r antibiotic deseq}
# Have to specify pairwise contrasts here.
AZTvNONE = results(deseqABonly, contrast = c("antibiotic_treatment", "AZT", "NONE"))
ERYvNONE = results(deseqABonly, contrast = c("antibiotic_treatment", "ERY", "NONE"))
SFXvNONE = results(deseqABonly, contrast = c("antibiotic_treatment", "SFX", "NONE"))
ALLvNONE = results(deseqABonly, contrast = c("antibiotic_treatment", "ALL", "NONE"))
# Set alpha cutoff for certainty
alpha = 0.05
# Extract results for which alpha is less than cutoff
significantAZT = AZTvNONE[which(AZTvNONE$padj < alpha),]
significantERY = ERYvNONE[which(ERYvNONE$padj < alpha),]
significantSFX = SFXvNONE[which(SFXvNONE$padj < alpha),]
significantALL = ALLvNONE[which(ALLvNONE$padj < alpha),]
# Cleaning up results by binding with phyloseq object's rownames
significantAZT = cbind(as(significantAZT, "data.frame"), 
                       as(tax_table(microbiome)[rownames(significantAZT),], "matrix"))
significantERY = cbind(as(significantERY, "data.frame"), 
                       as(tax_table(microbiome)[rownames(significantERY),], "matrix"))
significantSFX = cbind(as(significantSFX, "data.frame"), 
                       as(tax_table(microbiome)[rownames(significantSFX),], "matrix"))
significantALL = cbind(as(significantALL, "data.frame"), 
                       as(tax_table(microbiome)[rownames(significantALL),], "matrix"))

# Visualizing
# AZT vs. NONE
# Changing ASVs with the same genus to unique
significantAZT$Genus <- as.character(significantAZT$Genus)
significantAZT[5,12] <- "Sphingomonas 1"
significantAZT[6,12] <- "Sphingomonas 2"
# Sorting by maximum fold change at the class taxonomic rank
xAZT = tapply(significantAZT$log2FoldChange, significantAZT$Genus, function(x) max(x))
xAZT = sort(xAZT, TRUE)
significantAZT$Genus = factor(as.character(significantAZT$Genus), levels = names(xAZT))
# Making figure
significantDiffAZTvNONE = ggplot(significantAZT, aes(x = Class, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("AZT") + coord_flip() +
  scale_fill_manual(values = pal) +
  theme(legend.position = "none", axis.ticks.y = element_blank())
# ERY vs. NONE
significantERY$Genus <- as.character(significantERY$Genus)
significantERY[c(2,3,5,6,8,10,13,14),12] <- c("Sphingopyxis 1", "Unidentified 1", "Unidentified 2", "Emticicia 1", "Unidentified 3", "Sphingopyxis 2", "Emticicia 2", "Unidentified 4")
xERY = tapply(significantERY$log2FoldChange, significantERY$Genus, function(x) max(x))
xERY = sort(xERY, TRUE)
significantERY$Genus = factor(as.character(significantERY$Genus), levels = names(xERY))
# Making figure
significantDiffERYvNONE = ggplot(significantERY, aes(x = Genus, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("ERY") + coord_flip() + 
  scale_fill_manual(values = pal) +
  theme(legend.position = "none", axis.ticks.y = element_blank())
# SFX vs. NONE
significantSFX$Genus <- as.character(significantSFX$Genus)
significantSFX[4,12] <- "Unidentified 1"
significantSFX[6,12] <- "Unidentified 2"
xSFX = tapply(significantSFX$log2FoldChange, significantSFX$Genus, function(x) max(x))
xSFX = sort(xSFX, TRUE)
significantSFX$Genus = factor(as.character(significantSFX$Genus), levels = names(xSFX))
# Making figure
significantDiffSFXvNONE = ggplot(significantSFX, aes(x = Genus, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("SFX") + coord_flip() + 
  scale_fill_manual(values = pal) +
  theme(legend.position = "none", axis.ticks.y = element_blank())
# ALL vs. NONE
significantALL$Genus <- as.character(significantALL$Genus)
significantALL[c(2,3,5,8,10,12,14,15,17),12] <- c("Sphingopyxis 1", "Unidentified 1", "Unidentified 2", "Unidentified 3", "Sphingopyxis 2", "Unidentified 4", "Unidentified 5", "Flavobacterium 1", "Flavobacterium 2")
xALL = tapply(significantALL$log2FoldChange, significantALL$Genus, function(x) max(x))
xALL = sort(xALL, TRUE)
significantALL$Genus = factor(as.character(significantALL$Genus), levels = names(xALL))
# Making figure
significantDiffALLvNONE = ggplot(significantALL, aes(x = Genus, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("ALL") + coord_flip() + 
  scale_fill_manual(values = pal) +
  theme(legend.position = "none", axis.ticks.y = element_blank())
for_DESeqAB_legend = ggplot(significantALL, aes(x = Genus, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("ALL") + coord_flip() + 
  scale_fill_manual(values = pal)
DESeq_legend <- get_legend(for_DESeqAB_legend)
# Arranging into one figure
antibioticDESeqFigure = plot_grid(significantDiffAZTvNONE, significantDiffERYvNONE, significantDiffSFXvNONE, significantDiffALLvNONE, align = "v")
# Saving image - will add legend in Adobe Illustrator
ggsave("antibiotic_DESeq.pdf", antibioticDESeqFigure, units = "in", height = 12, width = 12, dpi = 300)
ggsave("antibiotic_DESeq_legend.pdf", DESeq_legend, units = "in", height = 12, width = 12, dpi = 300)
```

Now looking at differences across temperatures for each antibiotic.
```{r deseq temp}
# Same steps as above, just contrasting temperature treatments now.
NONEtemp = results(deseqTEMP.NONE, contrast = c("temp_treatment", "11C", "18C"))
AZTtemp = results(deseqTEMP.AZT, contrast = c("temp_treatment", "11C", "18C"))
ERYtemp = results(deseqTEMP.ERY, contrast = c("temp_treatment", "11C", "18C"))
SFXtemp = results(deseqTEMP.SFX, contrast = c("temp_treatment", "11C", "18C"))
ALLtemp = results(deseqTEMP.ALL, contrast = c("temp_treatment", "11C", "18C"))
significantNONEtemp = NONEtemp[which(NONEtemp$padj < alpha),]
significantAZTtemp = AZTtemp[which(AZTtemp$padj < alpha),]
significantERYtemp = ERYtemp[which(ERYtemp$padj < alpha),]
significantSFXtemp = SFXtemp[which(SFXtemp$padj < alpha),]
significantALLtemp = ALLtemp[which(ALLtemp$padj < alpha),]
significantNONEtemp = cbind(as(significantNONEtemp, "data.frame"), 
                       as(tax_table(microbiome)[rownames(significantNONEtemp),], "matrix"))
significantAZTtemp = cbind(as(significantAZTtemp, "data.frame"), 
                       as(tax_table(microbiome)[rownames(significantAZTtemp),], "matrix"))
significantERYtemp = cbind(as(significantERYtemp, "data.frame"), 
                       as(tax_table(microbiome)[rownames(significantERYtemp),], "matrix"))
significantSFXtemp = cbind(as(significantSFXtemp, "data.frame"), 
                       as(tax_table(microbiome)[rownames(significantSFXtemp),], "matrix"))
significantALLtemp = cbind(as(significantALLtemp, "data.frame"), 
                       as(tax_table(microbiome)[rownames(significantALLtemp),], "matrix"))

# Visualizing
# NONE
# Changing ASVs with the same genus to unique
significantNONEtemp$Genus <- as.character(significantNONEtemp$Genus)
significantNONEtemp[1,12] <- "Unidentified"
# Sorting by maximum fold change at the class taxonomic rank
xNONEtemp = tapply(significantNONEtemp$log2FoldChange, significantNONEtemp$Genus, function(x) max(x))
xNONEtemp = sort(xNONEtemp, TRUE)
significantNONEtemp$Genus = factor(as.character(significantNONEtemp$Genus), levels = names(xNONEtemp))
# Making figure
significantDiffNONEtemp = ggplot(significantNONEtemp, aes(x = Genus, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("NONE") + coord_flip() +
  scale_fill_manual(values = pal) +
  theme(legend.position = "none", axis.ticks.y = element_blank())
# AZT
significantAZTtemp$Genus <- as.character(significantAZTtemp$Genus)
significantAZTtemp$Genus
significantAZTtemp[c(1,2,4,5),12] <- c("Unidentified 1", "Unidentified 2", "Unidentified 3", "Unidentified 4")
# Sorting by maximum fold change at the class taxonomic rank
xAZTtemp = tapply(significantAZTtemp$log2FoldChange, significantAZTtemp$Genus, function(x) max(x))
xAZTtemp = sort(xAZTtemp, TRUE)
significantAZTtemp$Genus = factor(as.character(significantAZTtemp$Genus), levels = names(xAZTtemp))
# Making figure
significantDiffAZTtemp = ggplot(significantAZTtemp, aes(x = Genus, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("AZT") + coord_flip() +
  scale_fill_manual(values = pal) +
  theme(legend.position = "none", axis.ticks.y = element_blank())
# ERY
significantERYtemp$Genus <- as.character(significantERYtemp$Genus)
significantERYtemp[c(2,3,8),12] <- c("Unidentified 1", "Unidentified 2", "Unidentified 3")
# Sorting by maximum fold change at the class taxonomic rank
xERYtemp = tapply(significantERYtemp$log2FoldChange, significantERYtemp$Genus, function(x) max(x))
xERYtemp = sort(xERYtemp, TRUE)
significantERYtemp$Genus = factor(as.character(significantERYtemp$Genus), levels = names(xERYtemp))
# Making figure
significantDiffERYtemp = ggplot(significantERYtemp, aes(x = Genus, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("ERY") + coord_flip() +
  scale_fill_manual(values = pal) +
  theme(legend.position = "none", axis.ticks.y = element_blank())
# SFX
significantSFXtemp$Genus <- as.character(significantSFXtemp$Genus)
significantSFXtemp[c(1,3,5,8,9,10,11,12),12] <- c("Unidentified 1", "Flavobacterium 1", "Unidentified 2", "Unidentified 3", "Unidentified 4", "Unidentified 5", "Flavobacterium 2", "Flavobacterium 3")
# Sorting by maximum fold change at the class taxonomic rank
xSFXtemp = tapply(significantSFXtemp$log2FoldChange, significantSFXtemp$Genus, function(x) max(x))
xSFXtemp = sort(xSFXtemp, TRUE)
significantSFXtemp$Genus = factor(as.character(significantSFXtemp$Genus), levels = names(xSFXtemp))
# Making figure
significantDiffSFXtemp = ggplot(significantSFXtemp, aes(x = Genus, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("SFX") + coord_flip() +
  scale_fill_manual(values = pal) +
  theme(legend.position = "none", axis.ticks.y = element_blank())
# ALL
significantALLtemp$Genus <- as.character(significantALLtemp$Genus)
# Sorting by maximum fold change at the class taxonomic rank
xALLtemp = tapply(significantALLtemp$log2FoldChange, significantALLtemp$Genus, function(x) max(x))
xALLtemp = sort(xALLtemp, TRUE)
significantALLtemp$Genus = factor(as.character(significantALLtemp$Genus), levels = names(xALLtemp))
# Making figure
significantDiffALLtemp = ggplot(significantALLtemp, aes(x = Genus, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("ALL") + coord_flip() +
  scale_fill_manual(values = pal) +
  theme(legend.position = "none", axis.ticks.y = element_blank())
# Legend
temp_plot_for_legend <- ggplot(significantSFXtemp, aes(x = Genus, y = log2FoldChange, fill = Class)) + 
  geom_bar(stat = "identity") + 
  ggtitle("SFX") + coord_flip() +
  scale_fill_manual(values = pal)
temp_legend <- get_legend(temp_plot_for_legend)

tempDESeqFigure = plot_grid(significantDiffNONEtemp, significantDiffAZTtemp, significantDiffERYtemp, significantDiffSFXtemp, significantDiffALLtemp, ncol = 3, nrow = 2)
# Saving image - will add legend in Adobe Illustrator
ggsave("temp_DESeq.pdf", tempDESeqFigure, units = "in", height = 12, width = 12, dpi = 300)
ggsave("temp_DESeq_legend.pdf", temp_legend, units = "in", height = 12, width = 12, dpi = 300)
```

Final statistics.
```{r all stats}
# Cumulative reproduction
summary(repro.anova)
tHSD
# Reproductive timing
summary(first.anova)
tHSD.first
# Body size/growth
summary(growth.anova)
tHSD.growth
# Survival
summary(fit)
# Microbiome composition across antibiotics
noneVaztResults
noneVeryResults
noneVsfxResults
noneVallResults
# Differentially abundant taxa across antibiotics
significantAZT
significantERY
significantSFX
significantALL
# Microbiome composition in each antibiotic across temperatures
NONEresults
AZTresults
ERYresults
SFXresults
ALLresults
# Differentially abundant taxa in each antibiotic across temperatures
significantNONEtemp
significantAZTtemp
significantERYtemp
significantSFXtemp
significantALLtemp
```

Final figures.
```{r all figures}
# Cumulative reproduction
cumulative_reproduction + ggtitle("Cumulative Reproduction")
# Reproductive timing
day_of_firstbrood + ggtitle("First Brood Day")
# Body size/growth
growth_by_treatment + ggtitle("Growth")
# Survival curves and hazard ratios
survival_hazard_ratios + ggtitle("Survival Hazard Ratios")
survival_curves + ggtitle("Survival Curves")
# The Daphnia microbiome in healthy adults at the class level and the 10 most abundant ASVs
# Saved using an arrangeGrob, so just replotting with cowplot
plot_grid(adultMicrobiomeGenera, adultMicrobiomeClass, nrow = 1)
# The Daphnia microbiome across antibiotics
microbiomeByAB
# Differentially abundant taxa across antibiotics
antibioticDESeqFigure
# The Daphnia microbiome in each antibiotic across temperatures
# Saved using an arrangeGrob, so just replotting with cowplot
plot_grid(NONEfig, AZTfig, ERYfig, SFXfig, ALLfig, nrow = 1)
# Differentially abundant taxa in each antibiotic across temperatures
tempDESeqFigure
```

